src/PostgREST/MediaType.hs

```diff
@@ -32,83 +49,95 @@ data MediaType
 toContentType :: MediaType -> Header
 toContentType ct = (hContentType, toMime ct <> charset)
   where
     charset = case ct of
-      MTOctetStream -> mempty
-      MTOther _     -> mempty
-      _             -> "; charset=utf-8"
+      MTNormal MTOctetStream -> mempty
+      MTNormal (MTOther _)   -> mempty
+      _                      -> "; charset=utf-8"
 
 -- | Convert from MediaType to a ByteString representing the mime type
 toMime :: MediaType -> ByteString
-toMime MTApplicationJSON = "application/json"
-toMime MTGeoJSON         = "application/geo+json"
-toMime MTTextCSV         = "text/csv"
-toMime MTTextPlain       = "text/plain"
-toMime MTTextXML         = "text/xml"
-toMime MTOpenAPI         = "application/openapi+json"
-toMime MTSingularJSON    = "application/vnd.pgrst.object+json"
-toMime MTUrlEncoded      = "application/x-www-form-urlencoded"
-toMime MTOctetStream     = "application/octet-stream"
-toMime MTAny             = "*/*"
-toMime (MTOther ct)      = ct
+toMime (MTNormal x) = toMimeNormal x
 toMime (MTPlan (MTPlanAttrs mt fmt opts)) =
   "application/vnd.pgrst.plan+" <> toMimePlanFormat fmt <>
-  (if isNothing mt then mempty else "; for=\"" <> toMime (fromJust mt) <> "\"") <>
+  (if isNothing mt then mempty else "; for=\"" <> toMimeNormal (fromJust mt) <> "\"") <>
   (if null opts then mempty else "; options=" <> BS.intercalate "|" (toMimePlanOption <$> opts))
 
+toMimeNormal :: NormalMedia -> ByteString
+toMimeNormal = \case
+  MTApplicationJSON -> "application/json"
+  MTGeoJSON         -> "application/geo+json"
+  MTTextCSV         -> "text/csv"
+  MTTextPlain       -> "text/plain"
+  MTTextXML         -> "text/xml"
+  MTOpenAPI         -> "application/openapi+json"
+  MTSingularJSON    -> "application/vnd.pgrst.object+json"
+  MTUrlEncoded      -> "application/x-www-form-urlencoded"
+  MTOctetStream     -> "application/octet-stream"
+  MTAny             -> "*/*"
+  (MTOther ct)      -> ct
+
 toMimePlanOption :: MTPlanOption -> ByteString
-toMimePlanOption PlanAnalyze  = "analyze"
-toMimePlanOption PlanVerbose  = "verbose"
-toMimePlanOption PlanSettings = "settings"
-toMimePlanOption PlanBuffers  = "buffers"
-toMimePlanOption PlanWAL      = "wal"
+toMimePlanOption = \case
+  PlanAnalyze  -> "analyze"
+  PlanVerbose  -> "verbose"
+  PlanSettings -> "settings"
+  PlanBuffers  -> "buffers"
+  PlanWAL      -> "wal"
 
 toMimePlanFormat :: MTPlanFormat -> ByteString
 toMimePlanFormat PlanJSON = "json"
 toMimePlanFormat PlanText = "text"
 
--- | Convert from ByteString to MediaType. Warning: discards MIME parameters
+-- | Convert from ByteString to MediaType.
+--
+-- >>> decodeMediaType "application/json"
+-- MTNormal MTApplicationJSON
+--
+-- >>> decodeMediaType "application/vnd.pgrst.plan;"
+-- MTPlan (MTPlanAttrs Nothing PlanText [])
+--
+-- >>> decodeMediaType "application/vnd.pgrst.plan;for=\"application/json\""
+-- MTPlan (MTPlanAttrs (Just MTApplicationJSON) PlanText [])
```

Comment:
Here I tried adding a space like so:

```suggestion
-- >>> decodeMediaType "application/vnd.pgrst.plan; for=\"application/json\""
-- MTPlan (MTPlanAttrs (Just MTApplicationJSON) PlanText [])
```

And the decoding failed to get `Just MTApplicationJSON` (got `Nothing` instead). This clearly works on the feature specs:

https://github.com/PostgREST/postgrest/blob/78821a8fe7990022bb9036c24321ab411c55a3b1/test/spec/Feature/Query/PlanSpec.hs#L119

I thought there must be something wrong with whitespace on the `doctests`. This would have been easy to prove with the repl. But https://github.com/PostgREST/postgrest/issues/2627 continues to be a problem.